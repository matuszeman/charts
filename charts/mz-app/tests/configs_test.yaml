suite: configs
templates:
  - configs.yaml
  - deployment.yaml
tests:
  - it: should match snapshot with values-ci.yaml
    values:
      - ../values-ci.yaml
    asserts:
      - matchSnapshot: {}
  - it: when fromConfigMap is specified, then do not generate managed ConfigMap and generate deployment volume
    values:
      - values/minimum.yaml
    set:
      configs:
        not-managed-cm:
          fromConfigMap: "{{ .Release.Name }}-not-managed-cm"
    asserts:
      - hasDocuments:
          count: 0
        template: configs.yaml
      - contains:
          path: spec.template.spec.volumes
          content:
            name: config-not-managed-cm
            configMap:
              name: RELEASE-NAME-not-managed-cm
        template: deployment.yaml

  - it: when fromConfigMap and content is specified, then do not generate managed ConfigMap (prefer fromConfigMap)
    values:
      - values/minimum.yaml
    set:
      configs:
        not-managed-cm:
          fromConfigMap: "{{ .Release.Name }}-not-managed-cm"
          content:
            example.txt: example content
    asserts:
      - hasDocuments:
          count: 0
        template: configs.yaml
  - it: when fromSecret is specified, then do not generate managed Secret, generate deployment volume
    values:
      - values/minimum.yaml
    set:
      configs:
        not-managed-secret:
          fromSecret: "{{ .Release.Name }}-not-managed-secret"
    asserts:
      - hasDocuments:
          count: 0
        template: configs.yaml
      - contains:
          path: spec.template.spec.volumes
          content:
            name: config-not-managed-secret
            secret:
              secretName: RELEASE-NAME-not-managed-secret
        template: deployment.yaml
  - it: when fromSecret and content is specified, then do not generate managed Secret (prefer fromSecret)
    values:
      - values/minimum.yaml
    set:
      configs:
        not-managed-secret:
          fromSecret: "{{ .Release.Name }}-not-managed-secret"
          content:
            example.txt: example content
    asserts:
      - hasDocuments:
          count: 0
        template: configs.yaml
  - it: when awsSecret is specified, then generate SecretStore and ExternalSecret
    values:
      - values/minimum.yaml
    set:
      configs:
        aws-secrets:
          awsSecret:
            arn: arn:aws:iam::123456789012:role/external-secrets
            refreshInterval: 1h
          content:
            example.txt: example content
    asserts:
      - matchSnapshot:
          template: configs.yaml
  - it: when awsSecret with versionId is specified, then set it to externalsecret
    values:
      - values/minimum.yaml
    set:
      configs:
        aws-secrets:
          awsSecret:
            arn: arn:aws:iam::123456789012:role/external-secrets
            versionId: a1b2c3d4-5678-90ab-cdef-EXAMPLE22222
    asserts:
      - equal:
          path: spec.dataFrom[0].extract.version
          value: uuid/a1b2c3d4-5678-90ab-cdef-EXAMPLE22222
        template: configs.yaml
        documentIndex: 0
  - it: when awsSecret with no versionId is specified, then do not set it to externalsecret
    values:
      - values/minimum.yaml
    set:
      configs:
        aws-secrets:
          awsSecret:
            arn: arn:aws:iam::123456789012:role/external-secrets
            #versionId: a1b2c3d4-5678-90ab-cdef-EXAMPLE22222
    asserts:
      - notExists:
          path: spec.dataFrom[0].extract.version
        template: configs.yaml
        documentIndex: 0