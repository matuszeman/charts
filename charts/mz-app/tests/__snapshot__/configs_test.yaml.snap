should match snapshot with values-ci.yaml:
  1: |
    apiVersion: v1
    data:
      file1: file1 content
      plaintextkey: sometext
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: mz-app-0.1.0
      name: RELEASE-NAME-resource-suffix-config-app
  2: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: mz-app-0.1.0
      name: RELEASE-NAME-resource-suffix-config-app-secrets
    spec:
      dataFrom:
        - extract:
            conversionStrategy: Default
            decodingStrategy: None
            key: arn:aws:iam::123456789012:role/external-secrets
            version: uuid/a1b2c3d4-5678-90ab-cdef-EXAMPLE22222
      refreshInterval: 1h
      secretStoreRef:
        kind: SecretStore
        name: RELEASE-NAME-resource-suffix-config-secretsmanager
      target:
        creationPolicy: Owner
        name: RELEASE-NAME-resource-suffix-config-app-secrets
        template:
          engineVersion: v2
          metadata:
            labels:
              app.kubernetes.io/instance: RELEASE-NAME
              app.kubernetes.io/managed-by: Helm
              app.kubernetes.io/name: app-name
              app.kubernetes.io/version: 0.1.0
              helm.sh/chart: mz-app-0.1.0
  3: |
    apiVersion: v1
    data:
      php.ini: |
        memory_limit = 256M
        max_execution_time = 180
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: mz-app-0.1.0
      name: RELEASE-NAME-resource-suffix-config-php
  4: |
    apiVersion: external-secrets.io/v1beta1
    kind: SecretStore
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: mz-app-0.1.0
      name: RELEASE-NAME-resource-suffix-config-secretsmanager
    spec:
      provider:
        aws:
          auth:
            jwt:
              serviceAccountRef:
                name: RELEASE-NAME-resource-suffix
          region: eu-central-1
          service: SecretsManager
  5: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: mz-app-0.1.0
      name: RELEASE-NAME-resource-suffix
    spec:
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: app-name
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: app-name
        spec:
          affinity:
            podAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                - labelSelector:
                    matchExpressions:
                      - key: security
                        operator: In
                        values:
                          - S1
                  topologyKey: topology.kubernetes.io/zone
            podAntiAffinity:
              preferredDuringSchedulingIgnoredDuringExecution:
                - podAffinityTerm:
                    labelSelector:
                      matchExpressions:
                        - key: security
                          operator: In
                          values:
                            - S2
                    topologyKey: topology.kubernetes.io/zone
                  weight: 100
          containers:
            - image: 1234567890.dkr.ecr.eu-central-1.amazonaws.com/idp-example-web-app-apache:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: apache
              name: apache
              ports:
                - containerPort: 80
                  name: apache
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: apache
              resources:
                limits:
                  cpu: 200m
                  memory: 256Mi
                requests:
                  cpu: 100m
                  memory: 128Mi
            - env:
                - name: configFromAwsSecret
                  valueFrom:
                    secretKeyRef:
                      key: somekey
                      name: RELEASE-NAME-resource-suffix-config-app-secrets
                - name: configFromConfigContent
                  valueFrom:
                    configMapKeyRef:
                      key: plaintextkey
                      name: RELEASE-NAME-resource-suffix-config-app
                - name: configFromConfigMap
                  valueFrom:
                    configMapKeyRef:
                      key: somekey
                      name: RELEASE-NAME-not-managed-cm
                - name: configFromSecret
                  valueFrom:
                    secretKeyRef:
                      key: somekey
                      name: RELEASE-NAME-not-managed-secret
                - name: configKeyTemplatedValue
                  valueFrom:
                    configMapKeyRef:
                      key: cm-key
                      name: app-config-RELEASE-NAME
                - name: configKeyValue
                  valueFrom:
                    configMapKeyRef:
                      key: cm-key
                      name: app-config
                - name: floatValue
                  value: "456.789"
                - name: intValue
                  value: "123"
                - name: secretKeyTemplatedValue
                  valueFrom:
                    secretKeyRef:
                      key: secret-key
                      name: app-secrets-RELEASE-NAME
                - name: secretKeyValue
                  valueFrom:
                    secretKeyRef:
                      key: some-key
                      name: app-secrets
                - name: stringValue
                  value: string
                - name: templatedValue
                  value: value-RELEASE-NAME
              image: 1234567890.dkr.ecr.eu-central-1.amazonaws.com/test-server:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: app
              name: app
              ports:
                - containerPort: 9000
                  name: app
                  protocol: TCP
                - containerPort: 9000
                  name: anotherport
              readinessProbe:
                httpGet:
                  path: /
                  port: app
              resources:
                limits:
                  cpu: 200m
                  memory: 256Mi
                requests:
                  cpu: 100m
                  memory: 128Mi
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /
                  port: app
                periodSeconds: 10
              volumeMounts:
                - mountPath: /app-config
                  name: config-app
                  readOnly: true
                - mountPath: /app-secrets
                  name: config-app-secrets
                  readOnly: true
                - mountPath: /tmp/example.txt
                  name: config-not-managed-cm
                  readOnly: true
                  subPath: example.txt
                - mountPath: /etc/php.ini
                  name: config-php
                  readOnly: true
                  subPath: php.ini
                - mountPath: /app/public/var
                  name: ephemeral-app-cache
                  readOnly: false
          securityContext:
            sysctls:
              - name: net.ipv4.ip_unprivileged_port_start
                value: "0"
          serviceAccountName: RELEASE-NAME-resource-suffix
          volumes:
            - configMap:
                name: RELEASE-NAME-resource-suffix-config-app
              name: config-app
            - name: config-app-secrets
              secret:
                secretName: RELEASE-NAME-resource-suffix-config-app-secrets
            - configMap:
                name: RELEASE-NAME-not-managed-cm
              name: config-not-managed-cm
            - name: config-not-managed-secret
              secret:
                secretName: RELEASE-NAME-not-managed-secret
            - configMap:
                name: RELEASE-NAME-resource-suffix-config-php
              name: config-php
            - emptyDir:
                sizeLimit: 10Mi
              name: ephemeral-app-cache
when awsSecret is specified, then generate SecretStore and ExternalSecret:
  1: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: mz-app-0.1.0
      name: RELEASE-NAME-config-aws-secrets
    spec:
      dataFrom:
        - extract:
            conversionStrategy: Default
            decodingStrategy: None
            key: arn:aws:iam::123456789012:role/external-secrets
      refreshInterval: 1h
      secretStoreRef:
        kind: SecretStore
        name: RELEASE-NAME-config-secretsmanager
      target:
        creationPolicy: Owner
        name: RELEASE-NAME-config-aws-secrets
        template:
          engineVersion: v2
          metadata:
            labels:
              app.kubernetes.io/instance: RELEASE-NAME
              app.kubernetes.io/managed-by: Helm
              app.kubernetes.io/name: app-name
              app.kubernetes.io/version: 0.1.0
              helm.sh/chart: mz-app-0.1.0
  2: |
    apiVersion: external-secrets.io/v1beta1
    kind: SecretStore
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: mz-app-0.1.0
      name: RELEASE-NAME-config-secretsmanager
    spec:
      provider:
        aws:
          auth:
            jwt:
              serviceAccountRef:
                name: RELEASE-NAME
          region: eu-central-1
          service: SecretsManager
  3: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: mz-app-0.1.0
      name: RELEASE-NAME
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: app-name
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: app-name
        spec:
          containers:
            - image: 1234567890.dkr.ecr.eu-central-1.amazonaws.com/test-server:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: app
              name: app
              ports:
                - containerPort: 80
                  name: app
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: app
              resources:
                limits:
                  cpu: 200m
                  memory: 256Mi
                requests:
                  cpu: 100m
                  memory: 128Mi
          securityContext:
            sysctls:
              - name: net.ipv4.ip_unprivileged_port_start
                value: "0"
          serviceAccountName: RELEASE-NAME
          volumes:
            - name: config-aws-secrets
              secret:
                secretName: RELEASE-NAME-config-aws-secrets
